src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -25,12 +24,14 @@ internal static class HandshakeHelpers
             (byte)'B', (byte)'0', (byte)'D', (byte)'C', (byte)'8', (byte)'5', (byte)'B', (byte)'1', (byte)'1'
     };
 
-    // Verify Method, Upgrade, Connection, version,  key, etc..
-    public static void GenerateResponseHeaders(string key, string? subProtocol, IHeaderDictionary headers)
+    public static void GenerateResponseHeaders(bool isHttp1, string key, string? subProtocol, IHeaderDictionary headers)
     {
-        headers.Connection = HeaderNames.Upgrade;
-        headers.Upgrade = Constants.Headers.UpgradeWebSocket;
-        headers.SecWebSocketAccept = CreateResponseKey(key);
+        if (isHttp1)
+        {
+            headers.Connection = HeaderNames.Upgrade;
+            headers.Upgrade = Constants.Headers.UpgradeWebSocket;
+            headers.SecWebSocketAccept = CreateResponseKey(key);
+        }
         if (!string.IsNullOrWhiteSpace(subProtocol))
         {
             headers.SecWebSocketProtocol = subProtocol;
```

Comment:
Does something in the servers implementations make sure this header is lowercase?

Reply:
The header name? Yeah, the HPack encoder does that.